File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/Week/EditWeekDto.java
package com.salesianos.triana.BeElite.dto.Week;

import com.salesianos.triana.BeElite.dto.Program.ProgramDto;
import com.salesianos.triana.BeElite.model.Program;
import com.salesianos.triana.BeElite.model.Week;
import com.salesianos.triana.BeElite.model.Composite_Ids.WeekId;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
public class EditWeekDto{
        @NotEmpty(message = "Week must have name")
        String week_name;
        String original_name;
        String description;
        @NotNull(message = "Week must have creation date and time.")
        LocalDateTime created_at;
        @NotNull(message = "Week must be part of a program.")
        ProgramDto program;
        Long week_number;
        public static Week toEntity(EditWeekDto editedWeek, UUID p_id){
                return Week.builder()
                        .id(WeekId.of(editedWeek.getWeek_number(), editedWeek.getWeek_name(), p_id))
                        .createdAt(editedWeek.created_at)
                        .description(editedWeek.description)
                        .build();
        }
}

File: frontend/be_elite/lib/models/Week/edit_week_dto.dart
import 'package:be_elite/models/Program/program_dto.dart';

class EditWeekDto {
  // ignore: non_constant_identifier_names
  String? week_name;
  // ignore: non_constant_identifier_names
  String? original_name;
  String? description;
  // ignore: non_constant_identifier_names
  String? created_at;
  ProgramDto? program;
  // ignore: non_constant_identifier_names
  int? week_number;

  EditWeekDto(
      {
      // ignore: non_constant_identifier_names
      this.week_name,
      // ignore: non_constant_identifier_names
      this.original_name,
      this.description,
      // ignore: non_constant_identifier_names
      this.created_at,
      this.program,
      // ignore: non_constant_identifier_names
      this.week_number});

  factory EditWeekDto.fromJson(Map<String, dynamic> json) => EditWeekDto(
      week_name: json['week_name'] as String?,
      original_name: json['original_name'] as String?,
      description: json['description'] as String?,
      created_at: json['created_at'] as String?,
      program: json['program'] == null
          ? null
          : ProgramDto.fromJson(json['program'] as Map<String, dynamic>),
      week_number: json['week_number']);

  Map<String, dynamic> toJson() => {
        'week_name': week_name,
        'original_name': original_name,
        'description': description,
        'created_at': created_at,
        'program': program?.toJson(),
        'week_number': week_number,
      };
}
